app/views/play/level/modal/CourseVictoryModal.coffee

```diff
@@ -36,15 +37,17 @@ module.exports = class CourseVictoryModal extends ModalView
     @playSound 'victory'
     @nextLevel = new Level()
     @nextAssessment = new Level()
-    nextLevelPromise = api.levels.fetchNextForCourse({
-      levelOriginalID: @level.get('original')
-      @courseInstanceID
-      @courseID
-      sessionID: @session.id
-    }).then ({ level, assessment }) =>
-      @nextLevel.set(level)
-      @nextAssessment.set(assessment)
-    @supermodel.trackPromise(nextLevelPromise)
+    # next level for voyager calculated in onLoaded since it needs @classroom
+    unless utils.voyagerCourseIDs.includes(@courseID)
+      nextLevelPromise = api.levels.fetchNextForCourse({
+        levelOriginalID: @level.get('original')
+        @courseInstanceID
+        @courseID
+        sessionID: @session.id
+      }).then ({ level, assessment }) =>
+        @nextLevel.set(level)
+        @nextAssessment.set(assessment)
+      @supermodel.trackPromise(nextLevelPromise)
```

Comment:
Interested in learning about what `trackPromise`  method is adding to this promise?

Reply:
It mostly adds the callbacks for success and failure to the promise and handles the loading of the page based on that. I don't know exactly how, but it calls `onLoaded` once all the tracked requests and promises are completed.

Reply:
Thanks for explaining :)